From 07a474da7895cb007db7f0f353c098d7ef88c283 Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:18:14 -0400
Subject: [net] netfilter: nf_tables: disabling table hooks always succeeds

Message-id: <20141021161814.22888.29914.stgit@localhost.localdomain>
Patchwork-id: 97953
O-Subject: [RHEL7 net PATCH 056/124] netfilter: nf_tables: disabling table hooks always succeeds
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: f75edf5e9c9773cf6d0bd9b8d50ead41b4984569

 commit f75edf5e9c9773cf6d0bd9b8d50ead41b4984569
 Author: Pablo Neira Ayuso <pablo@netfilter.org>
 Date:   Sun Mar 30 14:04:52 2014 +0200

    netfilter: nf_tables: disabling table hooks always succeeds

    nf_tables_table_disable() always succeeds, make this function void.

    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netfilter/nf_tables_api.c b/net/netfilter/nf_tables_api.c
index e7ecfb3..040123c 100644
--- a/net/netfilter/nf_tables_api.c
+++ b/net/netfilter/nf_tables_api.c
@@ -381,7 +381,7 @@ err:
  return err;
 }
 
-static int nf_tables_table_disable(const struct nft_af_info *afi,
+static void nf_tables_table_disable(const struct nft_af_info *afi,
        struct nft_table *table)
 {
  struct nft_chain *chain;
@@ -391,8 +391,6 @@ static int nf_tables_table_disable(const struct nft_af_info *afi,
    nf_unregister_hooks(nft_base_chain(chain)->ops,
          afi->nops);
  }
-
- return 0;
 }
 
 static int nf_tables_updtable(struct sock *nlsk, struct sk_buff *skb,
@@ -412,9 +410,8 @@ static int nf_tables_updtable(struct sock *nlsk, struct sk_buff *skb,
 
   if ((flags & NFT_TABLE_F_DORMANT) &&
       !(table->flags & NFT_TABLE_F_DORMANT)) {
-   ret = nf_tables_table_disable(afi, table);
-   if (ret >= 0)
-    table->flags |= NFT_TABLE_F_DORMANT;
+   nf_tables_table_disable(afi, table);
+   table->flags |= NFT_TABLE_F_DORMANT;
   } else if (!(flags & NFT_TABLE_F_DORMANT) &&
       table->flags & NFT_TABLE_F_DORMANT) {
    ret = nf_tables_table_enable(afi, table);
-- 
1.7.1